modules/auxiliary/scanner/smb/smb_enumshares.rb

```diff
@@ -217,84 +172,102 @@ def get_files_info(ip, _rport, shares, info)
 
     logdata = ''
 
-    list = shares.collect { |e| e[0] }
-    list.each do |x|
-      x = x.strip
-      if (x == 'ADMIN$') || (x == 'IPC$')
+    shares.each do |share|
+      share_name = share[:name].strip
+      if SKIPPABLE_SHARES.include? share_name
+        next
+      end
+
+      if (share_name == USERS_SHARE) && !datastore['SpiderProfiles']
         next
       end
 
       if !datastore['ShowFiles']
-        print_status("Spidering #{x}.")
+        print_status("Spidering #{share_name}.")
+      end
+
+      begin
+        tree = simple.client.tree_connect("\\\\#{ip}\\#{share_name}")
+      rescue RubySMB::Error::UnexpectedStatusCode => e
+        print_error "Error when trying to connect to share #{share_name} - #{e.status_code.name}"
+        print_status "Spidering #{share_name} complete."
+        next
       end
+
       subdirs = ['']
-      if (x.strip == 'C$') && datastore['SpiderProfiles']
-        subdirs = profile_options(ip, x)
+      if (DEFAULT_SHARES.include? share_name) && datastore['SpiderProfiles']
+        subdirs = profile_options(tree, share)
       end
       until subdirs.empty?
-        depth = subdirs[0].count('\\')
-        if datastore['SpiderProfiles'] && (x == 'C$')
-          if depth - 2 > datastore['MaxDepth']
+        depth = subdirs.first.count('\\')
+
+        if DEFAULT_SHARES.include? share_name
+          if datastore['SpiderProfiles']
+            if (depth - 2) > datastore['MaxDepth']
```

Comment:
I don't like this personally as it goes against RuboCop standards but I can't find a way right now to simplify this without breaking the way the code works.

Reply:
Fixed in latest commit 👍

Reply:
Verified resolved in https://github.com/rapid7/metasploit-framework/pull/15854/commits/1a435f8a774817d4a328d542c831ee0b63a398a0, thanks for fixing this!